packages/contracts-bedrock/scripts/DeploySuperchain.s.sol

```diff
@@ -123,114 +124,119 @@ contract DeploySuperchainInput is CommonBase {
-    // The fields must be in alphabetical order for vm.parseToml to work.
-    struct Output {
-        ProtocolVersions protocolVersionsImpl;
-        ProtocolVersions protocolVersionsProxy;
-        SuperchainConfig superchainConfigImpl;
-        SuperchainConfig superchainConfigProxy;
-        ProxyAdmin superchainProxyAdmin;
-    }
-
-    // We use a similar pattern as the input contract to expose outputs. Because outputs are set
-    // individually, and deployment steps are modular and composable, we do not have an equivalent
-    // to the overall `inputSet` variable. However, we do hold everything in a struct, then
-    // similarly expose each field via a getter method. This getter method reverts if the output has
-    // not been set, ensuring that the caller cannot access any output fields until they have been set.
-    Output internal outputs;
+    // All outputs are stored in storage individually, with the same rationale as doing so for
+    // inputs, and the same pattern is used below to expose the outputs.
+    ProtocolVersions internal _protocolVersionsImpl;
+    ProtocolVersions internal _protocolVersionsProxy;
+    SuperchainConfig internal _superchainConfigImpl;
+    SuperchainConfig internal _superchainConfigProxy;
+    ProxyAdmin internal _superchainProxyAdmin;
 
     // This method lets each field be set individually. The selector of an output's getter method
     // is used to determine which field to set.
     function set(bytes4 sel, address _address) public {
-        if (sel == this.superchainProxyAdmin.selector) outputs.superchainProxyAdmin = ProxyAdmin(_address);
-        else if (sel == this.superchainConfigImpl.selector) outputs.superchainConfigImpl = SuperchainConfig(_address);
-        else if (sel == this.superchainConfigProxy.selector) outputs.superchainConfigProxy = SuperchainConfig(_address);
-        else if (sel == this.protocolVersionsImpl.selector) outputs.protocolVersionsImpl = ProtocolVersions(_address);
-        else if (sel == this.protocolVersionsProxy.selector) outputs.protocolVersionsProxy = ProtocolVersions(_address);
+        if (sel == this.superchainProxyAdmin.selector) _superchainProxyAdmin = ProxyAdmin(_address);
+        else if (sel == this.superchainConfigImpl.selector) _superchainConfigImpl = SuperchainConfig(_address);
+        else if (sel == this.superchainConfigProxy.selector) _superchainConfigProxy = SuperchainConfig(_address);
+        else if (sel == this.protocolVersionsImpl.selector) _protocolVersionsImpl = ProtocolVersions(_address);
+        else if (sel == this.protocolVersionsProxy.selector) _protocolVersionsProxy = ProtocolVersions(_address);
         else revert("DeploySuperchainOutput: unknown selector");
     }
 
     // Save the output to a TOML file.
+    // We fetch the output values using external calls to the getters to verify that all outputs are
+    // set correctly before writing them to the file.
     function writeOutputFile(string memory _outfile) public {
```

Comment:
is there somewhere I can see what this output file looks like?

Reply:
We don't (yet?) have a fixture for this, I would just run `forge script scripts/DeploySuperchain.s.sol:DeploySuperchain "run(string,string)" "test/fixtures/test-deploy-superchain.toml" "out.toml"`

Reply:
This command doesn't work for me. I added `--sig` and I get a different error: 
```
forge script scripts/DeploySuperchain.s.sol:DeploySuperchain --sig "run(string,string)" "test/fixtures/test-deploy-superchain-in.toml" "out.toml"
```

error:

```
    │   ├─ [0] VM::writeToml("<stringified TOML>", "./out.toml")
    │   │   └─ ← [Revert] vm.writeToml: the path out.toml is not allowed to be accessed for write operations
    │   └─ ← [Revert] vm.writeToml: the path out.toml is not allowed to be accessed for write operations
    └─ ← [Revert] vm.writeToml: the path out.toml is not allowed to be accessed for write operations
```

Reply:
@blmalone you could maybe write to `.testdata/out.toml` as a workaround for now as forge has [fs permissions](https://github.com/ethereum-optimism/optimism/blob/opsm/remove-structs-DeploySuperchain/packages/contracts-bedrock/foundry.toml#L51) for that dir. 

@mds1 I'm actually getting another error that I think probably precedes Blaine's.

```
script failed: vm.readFile: failed to read from "/Users/maurelian/projects/o/mono/packages/contracts-bedrock/test/fixtures/test-deploy-superchain.toml": No such file or directory (os error 2)
```

Reply:
Ahh, I see, it's `test-deploy-superchain-in.toml`

Reply:
Looks like everything is worked out here, so resolving this